core/src/main/scala/org/apache/spark/internal/config/Kafka.scala

```diff
@@ -79,4 +79,13 @@ private[spark] object Kafka {
         "For further details please see kafka documentation. Only used to obtain delegation token.")
       .stringConf
       .createOptional
+
+  val TOKEN_SASL_MECHANISM =
+    ConfigBuilder("spark.kafka.token.sasl.mechanism")
```

Comment:
Maybe `spark.kafka.token.sasl.mechanism` -> `spark.kafka.sasl.token.mechanism`?

Reply:
The `token` keyword is there for a reason (maybe it can be changed to reflect the meaning better).

All in all there are 2 `sasl.mechanism` used:
* To obtain token, which is hardcoded to `GSSAPI` at the moment without user parameter
* To authenticate the obtained token, which is this

To be consistent with the actual naming the first one would be `spark.kafka.sasl.token.mechanism` and the second something else.

All in all if we think the naming structure can be changed.

Reply:
The other driver what I've considered is to have parameter names in spark which can be searched in kafka's documentation easily:
* Prefix: `spark.kafka` which means inside spark a kafka parameter
* Parameter name: `sasl.mechanism` which can be searched easily on [Kafka configuration page](https://kafka.apache.org/documentation/#configuration)

In order not to break parameter names apart I've just added a parameter prefix this case.

Reply:
I don't think `token.sasl` is the correct hierarchy. SASL uses tokens.

Reply:
Ok, will change.